internal/controller/ingress/httpsedge_controller.go

```diff
@@ -349,6 +353,28 @@ func (r *HTTPSEdgeReconciler) setEdgeTLSTermination(ctx context.Context, edge *n
 	return err
 }
 
+func (r *HTTPSEdgeReconciler) setEdgeMutualTLS(ctx context.Context, edge *ngrok.HTTPSEdge, mtls *ingressv1alpha1.EndpointMutualTLS) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	client := r.NgrokClientset.EdgeModules().HTTPS().MutualTLS()
+	if mtls == nil {
+		if edge.MutualTls == nil {
+			log.V(1).Info("Edge Mutual TLS matches spec")
```

Comment:
Should this `return nil` like `setEdgeTLSTermination` does? Looks like this will move forward with a delete of what it thinks isn't there.

Reply:
You are right, it should have an early return if both are `nil`. I'll fix it. I don't think I caught it in my testing because if the desired state is no mTLS, I think it may still be valid to delete the mTLS config from the edge. Either way, I'll fix it. Thanks @bobzilladev.